packages/ec-core/src/distributions/collection.rs

```diff
@@ -1,92 +1,194 @@
-/// This module contains the implementation of the `CollectionGenerator` struct
-/// and related traits and functions.
+use std::collections::{LinkedList, VecDeque};
+
+use rand::prelude::Distribution;
+
+/// [`Distribution`] adapter that turns a [`Distribution`] of elements `T` into
+/// a [`Distribution`] of collections of elements (i.e. `Vec<T>`) of a given
+/// length.
 ///
-/// The `CollectionGenerator` struct is used to generate a collection of random
-/// elements. It takes an element generator and a size as input, and generates a
-/// `Vec` of random elements based on the specified size and the mechanism for
-/// generating random elements.
+/// Sampling from this distribution samples from the element collection `length`
+/// times.
 ///
-/// The module also defines the `ConvertToCollectionGenerator` trait, which
-/// provides methods for converting a type into a `CollectionGenerator`. This
-/// trait is implemented for any type that implements the `Generator` trait.
+/// # Provided conversion methods
 ///
-/// Finally, the module implements the `Generator` trait for the
-/// `CollectionGenerator` struct, allowing it to generate a `Vec` of random
-/// elements using the `generate` method.
-use rand::prelude::Distribution;
-
-/// Information for generating a collection of random elements.
+/// For ease of use, the trait [`ConvertToCollectionDistribution`] provides
+/// chainable constructors on any [`Distribution`] to turn that distribution
+/// into a collection distribution.
+///
+/// Those constructors are:
+/// - [`ConvertToCollectionDistribution::into_collection`], a owning
+///   constructor, moving the child distribution
+/// - [`ConvertToCollectionDistribution::to_collection`], a borrowing
+///   constructor, borrowing the child distribution.
+///
+/// Usage:
+///
+/// ```
+/// # use rand::distr::StandardUniform;
+/// # use ec_core::distributions::collection::{Collection, ConvertToCollectionDistribution};
+/// #
+/// let collection_distribution: Collection<StandardUniform> = StandardUniform.into_collection(10);
```

Comment:
Not a huge deal, but it would be nice if it were clearer than `10` is the size/length of the generated/sampled collection. Maybe just give it a name?

```rust
let collection_size = 10;
/// ...
let collection_distribution: Collection<StandardUniformL> 
    = StandardUniform.into_collection(collection_size);
```